app/modules/inspect/containers/configuration.jsx

```diff
@@ -0,0 +1,245 @@
+/**
+ * Copyright (c) 2014-present PlatformIO <contact@platformio.org>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Button, Col, Form, Row, Select, Switch } from 'antd';
+import { inspectProject, saveConfiguration } from '@inspect/actions';
+
+import PropTypes from 'prop-types';
+import React from 'react';
+import { connect } from 'react-redux';
+import { loadProjects } from '@project/actions';
+import { selectProjects } from '@project/selectors';
+import { selectSavedConfiguration } from '@inspect/selectors';
+
+class InspectionFormComponent extends React.Component {
+  static propTypes = {
+    form: PropTypes.object,
+    inspectProject: PropTypes.func.isRequired,
+    loadProjects: PropTypes.func.isRequired,
+    projects: PropTypes.arrayOf(
+      PropTypes.shape({
+        name: PropTypes.string.isRequired,
+        path: PropTypes.string.isRequired
+      })
+    ),
+    savedConfiguration: PropTypes.object,
+    saveConfiguration: PropTypes.func.isRequired
+  };
+
+  constructor(...args) {
+    super(...args);
+    this.state = {};
+  }
+
+  componentDidMount() {
+    this._isMounted = true;
+    const defaults = {
+      memory: true,
+      code: true
+    };
+    this.props.loadProjects();
+    this.props.form.setFieldsValue(this.props.savedConfiguration || defaults);
+  }
+
+  componentWillUnmount() {
+    this._isMounted = false;
+  }
+
+  isValid() {
+    const { projectDir, env, memory, code } = this.props.form.getFieldsValue();
+    return projectDir && projectDir.length && env && env.length && (memory || code);
+  }
+
+  handleSubmit = e => {
```

Comment:
Why we can't use original function syntax? 99% of this project use NAME(args)  {} but staring with PIO Inspect we have in file mix with "isValid() {" , "handleSubmit = e => {".

Let's use a unified format for all functions.